x-pack/plugins/beats_management/public/pages/main/tags.tsx

```diff
@@ -68,6 +77,31 @@ export class TagsPage extends React.PureComponent<TagsPageProps, TagsPageState>
     );
   }
 
+  private handleTagsAction = async (action: AssignmentActionType, payload: any) => {
+    switch (action) {
+      case AssignmentActionType.Delete:
+        const tags = this.getSelectedTags().map((tag: BeatTag) => tag.id);
+        const success = await this.props.libs.tags.delete(tags);
+        if (!success) {
+          alert(
+            'Some of these tags might be assigned to beats. Please ensure tags being removed are not activly assigned'
```

Comment:
It may be useful to add a column to the Tag list table that shows the number of associated beats, it will help the user avoid trying to delete tags that can't be removed.

Better yet, it may be useful to make the row un-selectable and provide a tooltip explaining why. Future improvement obviously.